pydantic/_internal/_std_types_schema.py

```diff
@@ -208,6 +208,9 @@ def path_schema_prepare_pydantic_annotations(
     import pathlib
 
     if source_type not in {
+        os.PathLike[bytes],
+        os.PathLike[str],
+        os.PathLike[Any],
```

Comment:
I'd rather us use `get_origin` on `source_type`, and if that's `os.PathLike`, then we can move forward with building a validator that takes into account how `os.PathLike` is parametrized